src/main/java/com/surrealdb/refactor/driver/parsing/ResultParser.java

```diff
@@ -0,0 +1,84 @@
+package com.surrealdb.refactor.driver.parsing;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.surrealdb.refactor.exception.UnhandledProtocolResponse;
+import com.surrealdb.refactor.types.QueryResult;
+import java.util.Map.Entry;
+import java.util.Set;
+
+public class ResultParser {
+    // this method parses the JSON element regardless of whether it is a single element
+    // or an array of elements
+
+    public ResultParser() {}
+
+    public QueryResult[] parseResultMessage(JsonElement resultMessage) {
```

Comment:
Tbh we can have this be JsonObject instead of JsonElement. The check should be done outside the function. It means we have less edge cases and less incorrect usage.

Reply:
I turned them into separate functions: 1 from JsonElement, 1 from JsonObject, and 1 from JsonArray